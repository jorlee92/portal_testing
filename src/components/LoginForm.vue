<template>
  <div>

    <div class="field">
      <label class="label" for="username">Username</label>
      <div class="control has-icons-left has-icons-right">
        <input class="input is-success" type="text" id="username" placeholder="username">
        <span class="icon is-small is-left">
          <i class="fas fa-user"></i>
        </span>
        <span class="icon is-small is-right">
          <i class="fas fa-check"></i>
        </span>
      </div>
      <p class="help is-success">This username is available</p>
    </div>

    <div class="field">
      <label class="label" for="password">Password</label>
      <div class="control">
        <input class="input" type="password" placeholder="password" id="password">
      </div>
    </div>

    <div class="field is-grouped">
      <div class="control">
        <button class="button is-link">Submit</button>
      </div>
      <div class="control">
        <button class="button is-text">Cancel</button>
      </div>
    </div>

  </div>
</template>

<script>
export default {
  data() {
    return {

    }
  },
}
</script>

<style lang="sass" scoped>
</style>