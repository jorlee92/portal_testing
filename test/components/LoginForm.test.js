import { shallow } from 'vue-test-utils'
import LoginForm from '@/components/LoginForm.vue'

describe('LoginForm', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(LoginForm)
  })
  it('should render the LoginForm correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should call attemptLogin when button is clicked', () => {
    const spy = jest.fn()
    wrapper.vm.attemptLogin = spy
    wrapper.update()
    const submitButton = wrapper.findAll('button').at(0)
    submitButton.trigger('click')
    expect(spy).toHaveBeenCalled()
  })
})
