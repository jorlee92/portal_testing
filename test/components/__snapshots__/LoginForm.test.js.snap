// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render the LoginForm correctly 1`] = `
<form>
  <div
    class="field"
  >
    <label
      class="label"
      for="username"
    >
      Username
    </label>
     
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        class="input"
        id="username"
        placeholder="username"
        type="text"
      />
       
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-user"
        />
      </span>
       
      <!---->
    </div>
     
    <p
      class="help is-success"
      style="display: none;"
    >
      This username is available
    </p>
  </div>
   
  <div
    class="field"
  >
    <label
      class="label"
      for="password"
    >
      Password
    </label>
     
    <div
      class="control"
    >
      <input
        class="input"
        id="password"
        placeholder="password"
        type="password"
      />
    </div>
  </div>
   
  <div
    class="field is-grouped"
  >
    <div
      class="control"
    >
      <button
        class="button is-link"
      >
        
        Submit
      
      </button>
    </div>
     
    <div
      class="control"
    >
      <button
        class="button is-text"
      >
        Cancel
      </button>
    </div>
  </div>
</form>
`;
